### PR TITLE
[Mobile app] - Refactoring authentication and onboarding screens

### PR DESCRIPTION
**Pull request by NoahNsimbe**
#### Summary of Changes (What does this PR do?)

- [x] Adds error handling during authentication.
- [x] Configures state management for onboarding screens.

#### Is this change ready to hit production in its current state?
- [x] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):

- [x] I consider this code done
- [ ] I've tested this locally
- [ ] This branch has been LGTM'ed by a reviewer

#### How should this be manually tested?
* Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?
- [MOB-124](https://airqoteam.atlassian.net/browse/MOB-124)

#### Screenshots (optional)

### CODE REVIEW

File: mobile/lib/blocs/auth_code/auth_code_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+    on<UpdateAuthCode>(_onUpdateAuthCode);
+    on<VerifyAuthCode>(_onVerifyAuthCode);
+    on<ResendAuthCode>(_onResendAuthCode);
+    on<InitializeAuthCodeState>(_onInitializeAuthCodeState);
+    on<ClearAuthCodeState>(_onClearAuthCodeState);
+    on<GuestUserEvent>(_onGuestUserEvent);
+    on<UpdateCountDown>(_updateCountDown);
+    on<UpdateVerificationId>(_onUpdateVerificationId);
+    on<UpdateEmailCredentials>(_onUpdateEmailCredentials);
+  }
```
Comment (NoahNsimbe) [inline mobile/lib/blocs/auth_code/auth_code_bloc.dart:25]: Authentication bloc. 
References:
https://pub.dev/packages/flutter_bloc
https://bloclibrary.dev/#/gettingstarted

File: mobile/lib/main_common.dart [hunk 2]
```diff
@@ -29,6 +29,18 @@ class AirQoApp extends StatelessWidget {
...
+        ),
+        BlocProvider(
+          create: (BuildContext context) => AuthCodeBloc(),
+        ),
+        BlocProvider(
+          create: (BuildContext context) => PhoneAuthBloc(),
+        ),
+        BlocProvider(
+          create: (BuildContext context) => EmailAuthBloc(),
+        ),
```
Comment (NoahNsimbe) [inline mobile/lib/main_common.dart:43]: Dependency injection to enable usage of a single bloc instance in multiple widgets.

File: mobile/lib/models/enum_constants.dart [hunk 3]
```diff
@@ -42,11 +42,134 @@ enum AnalyticsEvent {
...
+enum BlocStatus {
+  initial,
+  editing,
+  processing,
+  error,
+  success,
+  accountPreDeletionSuccess,
+}
+
+enum AuthenticationError {
```
Comment (NoahNsimbe) [inline mobile/lib/models/enum_constants.dart:111]: Authentication error messages

File: mobile/lib/screens/profile/profile_widgets.dart [hunk 4]
```diff
@@ -638,12 +634,16 @@ class EditCredentialsField extends StatelessWidget {
...
+              // ),
               GestureDetector(
-                onTap: () => _updateCredentials(context),
-                child: const EditCredentialsIcon(),
+                onTap: () {
+                  //TODO: implement re authentication
+                  // _updateCredentials(context);
+                },
+                child: Container(),
+                // child: const EditCredentialsIcon(),
```
Comment (NoahNsimbe) [inline mobile/lib/screens/profile/profile_widgets.dart:646]: Temporarily disabling change of login credentials.

**Review by Baalmart — APPROVED** [review]
Thanks @NoahNsimbe , this looks good to me.

**Pull request merged**